checks/raw/pinned_dependencies.go

```diff
@@ -273,8 +272,9 @@ var validateDockerfilesPinning fileparser.DoWhileTrueOnFileContent = func(
 						EndOffset: uint(child.EndLine),
 						Snippet:   child.Original,
 					},
-					Name:     asPointer(name),
-					PinnedAt: asPointer(asName),
+					Name:     asStringPointer(name),
+					PinnedAt: asStringPointer(asName),
+					Pinned:   asBoolPointer(pinned || regex.MatchString(name)),
```

Comment:
this duplicates some of the logic above. can we just use the value of `pinnedAsNames[asName]`?

Reply:
Yes, sure, nice catch!

Reply:
Doing a better analysis, here yes, we can replace `pinned || regex.MatchString(name)` by `pinnedAsNames[asName]`.

We have 3 pinned cases:
- `FROM something@sha256` - Pinned directly
- `FROM something@sha256 as base` - Pinned directly but defining AS NAME
- `FROM base` - Pinned indirectly using already defined AS NAME

In the code, we hold the info if the image is pinned directly or indirectly. `pinned` tells us if this image was already pinned AS NAME before, and `regex.MatchString(name)` tells us if the image is pinned right now.

In this part of the code, we handle AS NAME cases, so the if the image is pinned right now or before AS NAME, this info will be present in `pinnedAsNames[asName]`.